### PR TITLE
Fixes ENYO-1641

### PR DESCRIPTION
**Pull request by ryanjduffy**
## Issue

We have an existing and 2 in-flight overlay implementations that need to be consolidated into a consistent API.
## Fix

Add samples for moonstone/Overlay

Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

### CODE REVIEW

File: src/moonstone-samples/lib/OverlaySample.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+				mixins: [Overlay.Text], overlayShowing: 'spotlight', overlayTitle: 'Technics', overlaySubtitle: '12'
+			},
+			{kind: GridListImageItem, source: 'http://lorempixel.com/300/300/people', caption: 'Clamped Title', ontap: 'itemTapped',
+				mixins: [Overlay.Text], overlayShowing: 'spotlight', overlayComponents: [
+					// overlayComponents provided by Overlay.Text can be overridden for custom
+					// behavior like clamping the text to 3 lines. If the overlay is not always
+					// visible, clamped must be set to 'always' because the dimensions cannot be
+					// accurately calculated when the ClampedText control isn't showing.
+					{kind: ClampedText, maxLines: 3, clamped: 'always', classes: 'moon-overlay-text-title', content: 'This is probably too long to put in a text overlay but let\'s say you wanted to anyway'},
+					{classes: 'moon-overlay-text-subtitle', content: '3 lines'},
```
Comment (Djspaceg) [inline src/moonstone-samples/lib/OverlaySample.js]: Trailing comma. (Last one, I promise)

Comment (ryanjduffy) [inline src/moonstone-samples/lib/OverlaySample.js]: trailing comma ... gumble grumble ... blake ... grumble ...

Comment (Djspaceg) [inline src/moonstone-samples/lib/OverlaySample.js]: I approved it before this comma was added. ;) I just figure it's better to catch them now than to have to go back later.

**Comment by aarontam** [comment]
Hi @ryanjduffy, any chance you can fix the merge conflict sometime today? If not, I'll take care of it!

**Comment by aarontam** [comment]
As a general question/comment, I noticed with the samples that involve loading an image from a remote URL (and with a somewhat slow connection right now), the delay in loading the image causes the overlay to be incorrectly positioned initially, until the entire image is loaded. I think we might want to consider more gracefully handling these cases as slow-loading thumbnails or images can be a common scenario. Thoughts on this?

**Comment by Djspaceg** [comment]
Sounds like a good use of our shiny new image placeholder support. :D 

But in general, Images without a height and width assigned will behave as you describe; the browser automatically resizes them post loading. Without knowing their final dimensions before we start loading them, I don't think there's much/anything we can do differently. That being said, in the TV scenario, this will be less common because we have local assets and/or the sizes are handled by our GridListImageItem which assigns the sizes itself. It's possible, but unlikely in production because there's preventative measures apps can take to avoid it.

**Comment by aarontam** [comment]
Yep that sounds good to me. :) I was thinking more about the scenario where developers will be copying this code verbatim - if we had some generally best practices for these cases, such as using the placeholder image (when it's merged) or specifying explicit dimensions, then we would at least cover these cases sufficiently.

Edit: Ah I see @ryanjduffy added those, thanks!

**Pull request merged**